### ISSUE TITLE
TB does not support directly assigning EXCEL range to arr arrays

### ISSUE THREAD

**Issue by forliny** [twinbasic/twinbasic#1598]
**Describe the bug**
TB does not support directly assigning EXCEL range to arr arrays

dim rag as  Range 
set rag=Range("a1:d5")
Dim Arr() As Variant
    Arr = rag   ---------------[LINKER] compilation (codegen) error detected in 'XXX' at line #5
 Set rag = Nothing

**Comment by FullValueRider** [comment]
That's likely because you haven't defined Range as an Excel range. twinBasic is like VB6 in that you need to create an appropriate application object.  Unlike VBA, where if you are using Excel VBA the application object is predefined.  This is why the terribly terribly bad practise of unqualified Range usage in Excel should be avoided.

**Comment by forliny** [comment]
不是的，肯定不是这个原因。实际上我使用的range对象都是指定的Workbook对象的某个sheets("XX")表的Range范围，上面的代码只是我简化出来的。

**Comment by fafalone** [comment]
Please post in English; this is an English-language forum and you'll be more likely to get help if people don't need to translate your comments.

**Comment by forliny** [comment]
I already know the reason for this, it is because tb did not obtain the default attribute value of range, and needs to use range.value to proceed

**Comment by GCuser99** [comment]
But does it not count as a bug? The code you posted originally does work fine in VBA...

**Comment by GCuser99** [comment]
Vaguely similar to these:
https://github.com/twinbasic/twinbasic/issues/1753
https://github.com/twinbasic/twinbasic/issues/1724

**Comment by fafalone** [comment]
It does seem fine in VBA so maybe leave this open.

**Comment by FullValueRider** [comment]
The following code does not run in  Word VBA.  Methinks that far too many people use the term VBA without qualifying the application to which they refer.

```VBA
Sub TestUnqualifiedRangeInWord()
    Dim rag As Range
    Set rag = Range("a1:d5")  ' VBA in Word highlights Range and refuses to run the macro.
    Dim Arr() As Variant
    Arr = rag 
    Set rag = Nothing
End Sub
```

**Comment by GCuser99** [comment]
The OP specified Excel as the host.

**Comment by FullValueRider** [comment]
> The OP specified Excel as the host.

Nope.  The OP specified that the Excel Range keyword would not work in TB.

**Comment by fafalone** [comment]
While there's some VB6-only guys like me here I think even we would understand the `Range` object was an Excel Object Library thing 😁

**Comment by FullValueRider** [comment]
Not at all to be confused with .Range object which is a property of a number of Word objects <big grin>.

**Comment by FullValueRider** [comment]
That should have ended with a  **Big grin**

**Comment by GCuser99** [comment]
Small hint in the title of the post, even without reading the OP text... 😁

**Comment by forliny** [comment]
I hope you can pay attention to the issues of Excel Add in UDF, taskpane, and host debugging,

 which are very important for Excel plugin development. 

I am very much looking forward to updates on this aspect

https://github.com/twinbasic/twinbasic/issues/1860
